tests/test_layout.py

```diff
@@ -0,0 +1,43 @@
+
+def explode_and_remove(hlwm,create_client,number_clients):
```

Comment:
I'm not really sure what's going on here, so I can't make concrete suggestion yet, but it looks like this test case could be much improved.

For unit tests, there's a proven pattern called *Arrange-Act-Assert*. It encourages to focus on a single thing being tested per case, by first *arranging* the preconditions for the doing the thing, then *acting* out the thing, then *asserting* the expected outcome. Those three phases are commonly separated by blank lines. Having a lot of lines per phase, or growing a desire for additional blank lines is usually a good indicator for a vaguely scoped test case or lack of a useful fixture.

It often leads to quite repetitive test code (see some of my recently added tests), so I know it's hard to live by. But ultimately, it yields much more maintainable and understandable tests, in my experience.

Reply:
Let me also note pytest's [parametrization feature](https://docs.pytest.org/en/latest/parametrize.html) - often, if you have very similar tests (which only differ in data/arguments/...), that can make things much nicer.

Reply:
I agree, this is way to long and it contains two tests. I can try to separate it and I'll definitely use the pytest parameter feature!

Reply:
This is *much* better already! (see my new comment, though)